clipper.go

```diff
@@ -208,6 +208,10 @@ func mergeSettings() {
 		settings.Logfile = config.Logfile
 	} else {
 		settings.Logfile = defaults.Logfile
+		if runtime.GOOS == "linux" {
+			configDir := pathByExpandingTildeInPath(filepath.Dir(defaults.Logfile.value))
+			os.MkdirAll(configDir, 0700)
+		}
```

Comment:
Looks good, but I think this is the wrong place to be executing side-effectful code; this function should really just be about merging settings.

Would you be able to move this down lower, like around [here](https://github.com/wincent/clipper/blob/43a0654982c6e88c544fa2c8515ce3d642f87784/clipper.go#L273-L274) right before we try to open the log file? If you do that, it will additionally take effect for log files which come in via an explicit flag or a config file, which might be a benefit too.

Reply:
Should be fixed in the latest commit